### PR TITLE
Add original xfstests testsuite

### PR DESCRIPTION
**Pull request by SeroSun**
This is the first version of xfstests, it use the latest xfstests testsuite from git.kernel.org
It has some thing to be improved, I add some dependency packages in my ibs to make testsuite install all right. And the log analys hasn't finished, it only upload the full log folder in this moment. 
http://147.2.207.102/tests/694

### CODE REVIEW

File: tests/qa_automation/kernel_xfstests.pm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+sub run() {
+    my $self = shift;
+    $self->system_login();
+    $self->prepare_repos();
+    # create test partition
+    assert_script_run("parted -l -m 2>&1");
+    my $cmd           = "parted -l -m 2>&1| awk -F \':\' \'{if(\$5 == \"xfs\") print \$1}\'";
+    my $partition_num = $self->qa_script_output($cmd, 10);
+    my $partition     = "/dev/vda" . $partition_num;
```
Comment (mitiao) [inline tests/qa_automation/kernel_xfstests.pm]: Looks like you are using existing /home as a XFS test partition, and just for your information, the sles12_minimal_base+sdk_create_hdd on o.s.d with TOGGLEHOME=1, therefore there is **no** XFS partition on published hdd, unless you create a hdd with XFS partition.

File: tests/qa_automation/kernel_xfstests.pm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    assert_script_run("fdisk -l " . $partition, 10);
```
Comment (mitiao) [inline tests/qa_automation/kernel_xfstests.pm]: Acturally you don't need add timeout here for assert_script_run if you can expect the command finish executing within 90s.

File: tests/qa_automation/kernel_xfstests.pm (hunk 1 above)
Comment (SeroSun) [inline tests/qa_automation/kernel_xfstests.pm]: @mitiao thank you for your information:) 
I need to create a hdd with home partition, just remove the TOGGLEHOME=1 setting.

File: tests/qa_automation/kernel_xfstests.pm (hunk 2 above)
Comment (SeroSun) [inline tests/qa_automation/kernel_xfstests.pm]: Thanks I made a commit for it. All the timeout lease than 90s in assert_script_run have been removed.

**Comment by mitiao** [comment]
please squash and rebase the commits into 1

**Comment by coolo** [comment]
Most tests of the xfstests suite won't run in this setup as it needs a 2nd disk. I think it's the wrong statement if we merge xfstests as this.

**Comment by SeroSun** [comment]
@coolo Don't worry, I'll not trigger this test until I add the 2nd partition. Actually, after 2nd disk added in it still have 15%-20% cases won't run(those case need more than 2 partition), but I think it'll be more acceptable then.

**Pull request merged**